src/mainwindow.cpp

```diff
@@ -697,6 +700,37 @@ void MainWindow::find()
     consoleTabulator->terminalHolder()->currentTerminal()->impl()->toggleShowSearchBar();
 }
 
+void MainWindow::sendHistoryToEditor()
+{
+    const QString fn = QStandardPaths::writableLocation(QStandardPaths::TempLocation) + QLatin1String("/qterminal.history.") + QString::number(QCoreApplication::applicationPid());
+    QFile file(fn);
+    if (!file.open(QIODevice::WriteOnly)) {
+        qDebug() << "Failed to open" << file.fileName() << "for writing";
+        return;
+    }
+    TermWidgetImpl *impl = consoleTabulator->terminalHolder()->currentTerminal()->impl();
+    impl->saveHistory(&file);
+    file.close();
+    Properties::EditorType type = Properties::Instance()->editorType;
+    QString editor = Properties::Instance()->editor;
+    if (type == Properties::Editor_Auto) {
+        type = Properties::Editor_GUI;
+        for (const char *ed : { "vim", "vi", "less", "more", "nano", "pico", "view", "ed" }) {
```

Comment:
I prefer to avoid another collection of terminal editors in qterminal. This should be done by query the system (e.g., the Terminal= key in XDG standards).